kernel/src/arch/x86/smp/smp.cpp

```diff
@@ -177,7 +177,171 @@ namespace smp
                     }
                 },
                 core_id);
-
+            proc::make_kthread_args(
```

Comment:
Move this logic into its own file (something like `arch/x86/drivers/keyboard/ps2(.cpp/.h)`). 
Also refactor the actual read code into a function, since polling should be replaced with interrupts once I finish interrupt controller (I/O APIC).

Reply:
Also you probably only want to poll on one core, so just add a check for it somewhere.